zjit/src/hir.rs

```diff
@@ -2793,7 +2792,7 @@ mod tests {
         assert_method_hir_with_opcode("test", YARVINSN_newarray, expect![[r#"
```

Comment:
I think this test is now wrong; instead of making an array of `a`, it makes an array of `self`